addon/controllers/gis-search-form.js

```diff
@@ -117,6 +167,93 @@ export default Ember.Controller.extend({
         fieldName: field
       });
       this.send('doSearch', req);
+    },
+
+    /**
+      Handles click on 'open map' button.
+
+      @method actions.goToMap
+      @param {Object} mapModel Model of the map to transition
+    */
+    goToMap(mapModel) {
+      // may be it should pass event.ctrlKey
+      this.transitToMap(mapModel);
+    },
+
+    /**
+      Handles click on 'open a map with metadata' button.
+
+      @method actions.goToMapWithMetadata
+    */
+    goToMapWithMetadata() {
+      let selectedRows = this.get('_selectedRows');
+      let metadataIds = Object.keys(selectedRows).filter(function (item) {
+        return Ember.get(selectedRows, item);
+      });
+      this.transitToMapWithMetadata(this.get('_selectedMap'), metadataIds);
+    },
+
+    /**
+      Handles click on row select checkbox.
+
+      @method actions.onRowSelect
+      @param {String} rowId Id of selected row.
+      @param {Object} options Event options. We condider options.checked.
+    */
+    onRowSelect(rowId, options) {
+      this.set(`_selectedRows.${rowId}`, options.checked);
+      this.notifyPropertyChange('_selectedRows');
+    },
+
+    /**
+      Clears selected rows.
+
+      @method actions._clearSelectedRows
+    */
+    _clearSelectedRows() {
+      this.set('_selectedRows', {});
+      this.notifyPropertyChange('_selectedRows');
+    }
+  },
+
+  /**
+    Handles transition to selected map.
+
+    @method transitToMap
+    @param {Object} mapModel Model of the map to transition
+  */
+  transitToMap(mapModel) {
+    let mapRoute = this.get('mapRouteName');
+    Ember.assert(`The parameter 'mapRouteName' shouldn't be empty!`, !Ember.isNone(mapRoute));
```

Comment:
use Ember.isBlank instead of isNone
isNone will return false for empty strings, array, objects and so on

Reply:
Fixed